docs/DeveloperGuide.md

```diff
@@ -6,32 +6,215 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Implemented Features
 
+Following a restucture to the project to follow a more Object Oriented structure, as well as to allow for conformity and easy editing of code, many of our commands will follow a similar structure of the following:
+
+1. User enters a command, which is given to a Command Handler
+2. The handler passes this to a corresponding Parser object for that command, which will split it into its necessary parts
+3. The Parser will create a corresponding Command object for that command, which will perform its necessary actions
+4. The Command object will return successfully to the Command Handler, and the output will be given to the user (output for the user is stored in a `Response` class)
+
+The command to add a project is one example of this structure:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
```

Comment:
I strongly suggest using plantuml, the image is hard to read with different font size, 
![image](https://user-images.githubusercontent.com/69450377/160964004-a7fffa4e-b066-481b-aa5f-6b6346426c34.png)
different arrows
![image](https://user-images.githubusercontent.com/69450377/160964074-f2ea2b95-9b37-413c-9aa1-03a32fe304e1.png)
floating returns
![image](https://user-images.githubusercontent.com/69450377/160964100-bd8c8247-fa34-4822-94da-f1ced7ee0bec.png)
lack of colour...

Really plantuml helps solve all of that, i suggest you try